distribution/lib/Standard/Visualization/0.0.0-dev/src/Table/Visualization.enso

```diff
@@ -211,9 +211,101 @@ make_json_for_table dataframe max_rows all_rows_count include_index_col is_db_ta
-    pairs                = [header, value_type, data, all_rows, has_index_col, links, ["data_quality_metrics", data_quality_metrics] ,["type", "Table"], child_label, requires_number_formatting]
+    pairs                = [header, value_type, ["data", data], all_rows, has_index_col, links, ["data_quality_metrics", data_quality_metrics] ,["type", "Table"], child_label, ["is_using_server_sort_and_filter", is_db_table.not], requires_number_formatting]
     JS_Object.from_pairs pairs
 
+## PRIVATE
+
+   gets the distinct values of a column, used for filter options 
+
+   Arguments:
+   - table: table containing columns 
+   - column_index: the index of the column to get the distinct values for
+
+get_distinct_values_for_column : Table -> Integer -> JS_Object
+get_distinct_values_for_column table column_index =
+    column = table.at column_index
+    values = column.to_vector.distinct
+    result = JS_Object.from_pairs [["distinct_vals", values]]
+    result.to_text
+
+## PRIVATE
+
+   applies the sort model to the table.
+
+   Arguments:
+   - x: table to sort
+   - sort_col_index_list: indexes of columns to sort
+   - sort_direction_list: list of sort directions 1 for ascending -1 for descending
+
+apply_sort_to_table : Table -> List -> List -> Table
+apply_sort_to_table table sort_col_index_list sort_direction_list =
+    parse_direction d = if d == -1 then Sort_Direction.Descending else Sort_Direction.Ascending
+    get_pattern i = (Sort_Column.Index (sort_col_index_list.get i) (parse_direction (sort_direction_list.get i)))
+    sort_pattern = ((0.up_to sort_col_index_list.length).map idx-> get_pattern idx)
+    table.sort sort_pattern
+
+## PRIVATE
+
+   applies one filter to the table.
+
+   Arguments:
+   - table: table to sort
+   - filter_col: the column index to filter
+   - filter_action: filter action
+   - filter_val: values to filter
+   - filter_to_val: if a Between is the filer action this is the other value to provide a range
+
+apply_single_filter_to_table : Table -> Text -> Filter_Condition -> Any -> Any -> Table
+apply_single_filter_to_table table filter_col filter_action filter_val filter_to_val = 
+    case filter_action of
+      Filter_Condition.Between _ _ _ -> table.filter (filter_col+1) (Filter_Condition.Between filter_val filter_to_val)
+      Filter_Condition.Is_In _ _ -> if filter_val.length > 0 then table.filter (filter_col+1) (Filter_Condition.Is_In filter_val) else table.filter filter_col (Filter_Condition.Is_In [""])
+      _ -> table.filter (filter_col+1) (filter_action (filter_val)) 
+
+## PRIVATE
+
+   applies the filter model to the table.
+
+   Arguments:
+   - table: table to sort
+   - filter_col: indexes of columns with sorts applies
+   - filter_action: list of filter actions 
+   - filter_val: values to filter
+
+apply_filter_to_table : Table -> Integer -> List -> List -> List -> List -> Table
+apply_filter_to_table table i filter_col filter_action filter_val filter_to_val = 
```

Comment:
Not all arguments are documented here. If `Arguments` is included, IMO it should list all arguments.